gazelle/python/resolve.go

```diff
@@ -114,6 +125,30 @@ func importSpecFromSrc(pythonProjectRoot, bzlPkg, src string) resolve.ImportSpec
 	}
 }
 
+func isProtoLibrary(r *rule.Rule) bool {
+	return r.Kind() == pyProtoLibraryKind
+}
+
+func importsProtoLibrary(cfg *pythonconfig.Config, r *rule.Rule) []resolve.ImportSpec {
+	specs := []resolve.ImportSpec{}
+
+	// Determine the root module and emit an import for that,
+	// i.e. for //foo:foo_py_pb2, we'd get foo.foo_pb2
+	protoRelAttr := r.PrivateAttr(protoRelKey)
+	protoSrcsAttr := r.PrivateAttr(protoSrcsKey)
+	if protoRelAttr == nil || protoSrcsAttr == nil {
+		return nil
+	}
+
+	protoRel := protoRelAttr.(string)
+	for _, protoSrc := range protoSrcsAttr.([]string) {
+		generatedPbFileName := strings.TrimSuffix(protoSrc, ".proto") + "_pb2.py"
```

Comment:
Future work: if the proto import is guarded with `if TYPE_CHECKING`, then only include the `_pb2.pyi` file in the `pyi_deps` target attribute.